### ISSUE TITLE
Questions about KGT Task

### ISSUE THREAD

**Issue by Bloodline-2000** [apoorvumang/kgt5#20]
Nice work. However, I have some questions about the link prediction task on Wikidata5m dataset.
With your training code, I train the model for 4M steps, and the Hits@1 accuracy is only 0.22, which is far lower than the 0.267 reported in your paper. The code I ran is `main_accelerate.py, ` and the command I used for training is:
`CUDA_VISIBLE_DEVICES=0 python3 main_accelerate.py \
--save_prefix wd5m-1gpu \
--model_size small --dataset wikidata5m \
--batch_size 64 --save_steps 5000 \
--loss_steps 500`
Can you provide more details about how you obtain such a result of 0.267? Thanks.

**Comment by apoorvumang** [comment]
Can you share how you evaluated the model? Did you apply filtering?

**Comment by apoorvumang** [comment]
Also, which aliases did you use for entities and relations? It is recommended to use the ones specified in #18

**Comment by Bloodline-2000** [comment]
Thanks for you reply. I evaluated the model using the code `[eval_accelerate.py](https://github.com/apoorvumang/kgt5/blob/main/eval_accelerate.py)`. I didn't apply filtering. Does filtering mean the result in Tab.13 in your paper? And the results in Tab.2 is also under filtering?  For the KGC dataset, I use the file you provided in the ReadME, so I think it used the aliases you specified.

**Comment by apoorvumang** [comment]
1. All the reported results use filtering (including Tab 2), where we remove the seen entities from predictions. Please see the TransE paper for more details about filtering, and to see how we did it you can refer to the this code in apoorv-dump branch (sorry for the bad state of the code): https://github.com/apoorvumang/kgt5/blob/apoorv-dump/filtered_eval_from_scores_wd5m_v3.ipynb 
2. The file provided for aliases in the README is unfortunately not the best. An updated alias list can be found in #18 , I will be updated the README soon to reflect this

**Comment by Bloodline-2000** [comment]
Thanks for your reply. Can you also provide some details of how many steps are required for the training of a best model for wikidata5m dataset?

**Comment by apoorvumang** [comment]
We observed improvements till at least 50 epochs (trained on 4 GPUs), but the gains are diminishing. At 4M steps 1 GPU you should have comparable numbers I think (close to 0.28 MRR, 0.25 hits@1, filtered eval).

**Comment by Bloodline-2000** [comment]
Thank you. And for the `eval_accelerate.py`, could you give some details about what did you set `num_predictions` to be for the wikidata5m dataset? The default value is only 1.

**Comment by Bloodline-2000** [comment]
Besides, in the `[filtered_eval_from_scores_wd5m_v3.ipynb]` file, you compute `filtering_entities`. I am wondering what it stands for and why we need to skip the predictions with `len(filtering_entities) <= 1 ` in the Hits/MRR computation? Thanks.

**Comment by Bloodline-2000** [comment]
Another questions is: will different alias make large difference? I used the dataset you provided in README to train for 4M steps and used the `filtered_eval_from_scores_wd5m_v3.ipynb` and `eval_accelerate.py` in  apoorv-dump branch to perform evaluation. The results are as follows:
`hits@1 0.12818877551020408`
`hits@3 0.18622448979591838`
`hits@10 0.2653061224489796`
`mrr 0.17208884745938363`
`0.029974489795918366 were <10 length preds array without answer`

**Comment by apoorvumang** [comment]
1. We used `num_predictions = 500` for results reported in paper, but 200-400 should also be 'good enough'
2. For calculating MRR we don't need to skip predictions with `len(filtering_entities) <= 1`. Please comment that out, that was in the notebook for the Table 13 ablation done in the paper.
3. Aliases made a significant difference - the v3 aliases boosted performance from 0.26 MRR to 0.3 MRR for us. 

The numbers you show are pretty low, since the eval code is not correct (its for Table 13 ablation). Could you try commenting out len(filtering_entities) lines, or try using `filtered_eval_from_scores_wd5m_v3_clean.ipynb` instead?

**Comment by Bloodline-2000** [comment]
Thanks so much for your reply. Commenting out the len(filtering_entities) lines works. For the source of wiki5m dataset, are you using the Transductive split from [this website](https://deepgraphlearning.github.io/project/wikidata5m)?

**Comment by apoorvumang** [comment]
Yes, we had originally downloaded the dataset from that website in early 2021 (https://deepgraphlearning.github.io/project/wikidata5m) . However, they seem to have changed the dataset slightly (without specifying the changes afaik), resulting in the different number of entities/relations in their current split vs what we have (828 vs 822 relations, 4.8M vs 4.5M entities). LibKGE (https://github.com/uma-pi1/kge) also uses the previously provided dataset of wikidata5m

**Comment by Bloodline-2000** [comment]
Thank you for your help! I also found that training in multi-gpu with the code `CUDA_VISIBLE_DEVICES=1,2,3,4,5,7 python3 -m torch.distributed.launch --nproc_per_node 6 --use_env ./main_accelerate.py \
--save_prefix wd5m-6gpu \
--model_size small --dataset wikidata5m \
--batch_size 64 --save_steps 5000 \
--loss_steps 500` results in a much lower accuracy. I am wondering for multiple gpus setting, do you follow exactly the same command or you adjust the learning rate?  Thanks.

**Comment by github-cqy** [comment]
@apoorvumang @Bloodline-2000  hi,apoorvumang and bloodline-2000
After watching your conversation, I tried to run filtered_ eval_ from_ scores_ wd5m_ V3 file, but I'm missing the path (scores/scores_wd5m_v3_test. pickle). Can you help me? Thank you!

**Comment by Bloodline-2000** [comment]
@github-cqy You need to run the eval_accelerate.py file in the new branch.

**Comment by Bloodline-2000** [comment]
@apoorvumang Hi, with the new alias, at 400M steps 1 gpu, I cannot even beat the performance of the initial alias, is that normal? Could you share the dataset file you are using?
